### DISCUSSION TITLE
A lot of db calls

### DISCUSSION THREAD

**Discussion by CSharpDummy** [zauberzeug/nicegui#3120] [Q&A]
I have a lot of db calls that I need to do from the ui. Should I create fastapi routes for them and then call them or is there a better way in nicegui?
The methods need to be accessible from pages of different users.

**Comment by falkoschindler** [comment]
Hi @CSharpDummy,
You don't need FastAPI routes. The UI (e.g. a `ui.button`) can simply call any function that does the database request.

**Reply by frankvp11** [reply]
I think the only reason you'd want to turn it into routes would be if you are trying to go into like a micro-services routes
